File: libs/jgdsl/src/gdsl/GdslException.java
package gdsl;

/**
 * The GdslException class is the base class for all Gdsl specific
 * exceptions.
 * 
 * @author Julian Kranz
 * 
 */
public class GdslException extends RuntimeException {
  private static final long serialVersionUID = 1L;

  public GdslException () {
  }

  public GdslException (String message) {
    super(message);
  }

  public GdslException (Throwable cause) {
    super(cause);
  }

  public GdslException (String message, Throwable cause) {
    super(message, cause);
  }

}

File: libs/cppgdsl_legacy/src/gdsl_exception.cpp
/*
 * gdsl_exception.cpp
 *
 *  Created on: May 27, 2014
 *      Author: Julian Kranz
 */

#include <cppgdsl/gdsl_exception.h>
#include <iostream>

gdsl::gdsl_exception::gdsl_exception(std::string cppgdsl_message, std::string gdsl_message) {
  this->cppgdsl_message = cppgdsl_message;
  this->gdsl_message = gdsl_message;
}

std::ostream &gdsl::operator <<(std::ostream &out, gdsl_exception &_this) {
  out << "gdsl exception, original gdsl message: " << _this.gdsl_message << ", cppgdsl message: "
      << _this.cppgdsl_message;
  return out;
}
